pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
Here's what I'm implementing. First, we compute the patch from original to modified supplied by the user. Then, we compute the divergence from original to live, ignoring additions. If the divergence contains any changes or deletions that the user has not also made (i.e., any changes or deletions that are not also in the patch), then we have a conflict. If not, we apply the patch. Thoughts?